dashi/bcl2fastq/index_summary.py

```diff
@@ -196,82 +194,80 @@ def create_pie_chart(run, pruned, total_clusters):
     )
 
 
-@app.callback(
-    [dep.Output("run_select", "value"), dep.Output("error", "displayed")],
-    [dep.Input("bcl2fastq_url", "pathname")],
-)
-def change_url(pathname):
-    """ Allows user to enter Run name in URL which will update dropdown automatically, and the graphs.
-        If User enters any value that's not a valid run an error box will pop up and return user to most recent run
-
-        Parameters:
-             pathname: user-requested path.
+def assign_callbacks(app: dash.Dash):
+    @app.callback(
+        [
+            dep.Output("known_index_bar", "figure"),
+            dep.Output("unknown_index_bar", "figure"),
+            dep.Output("known_unknown_pie", "figure"),
+            dep.Output("known_fraction", "value"),
+        ],
+        [dep.Input("run_select", "value")],
+    )
+    def update_layout(run_alias):
+        """
+        When input(run dropdown) is changed, known index bar, unknown index bar,
+        piechart and textarea are updated
 
-        Returns:
-            The string value (without '/') of the user input for the drop-down to use
-            Error pop-up displayed depending on user input.
-    """
-    #   In a pathname, it automatically adds '/' to the beginning of the input even if pathname blank
-    #   While page loads, pathname is set to 'None'. Once page is loaded pathname is set to user input.
-    if pathname == "/" or pathname is None:
-        return all_runs[0], False
-    elif pathname[1:-2] not in all_runs:
-        return all_runs[0], True
-    else:
-        return pathname[1:-2], False
-
-
-@app.callback(
-    [
-        dep.Output("known_index_bar", "figure"),
-        dep.Output("unknown_index_bar", "figure"),
-        dep.Output("known_unknown_pie", "figure"),
-        dep.Output("known_fraction", "value"),
-    ],
-    [dep.Input("run_select", "value")],
-)
-def update_layout(run_alias):
-    """ When input(run dropdown) is changed, known index bar, unknown index bar, piechart and textarea are updated
         Parameter:
             run_alias: user-selected run name from dropdown
         Returns:
-            functions update_known_index_bar, update_unknown_index_bar, update_pie_chart's data value,
-            and update_pie_chart's fraction value
-    """
+            functions update_known_index_bar, update_unknown_index_bar,
+            update_pie_chart's data value, and update_pie_chart's fraction value
+        """
+
+        run = index[index[index_col.Run] == run_alias]
+        run = run[run[index_col.ReadNumber] == 1]
+        run = run[~run[index_col.SampleID].isna()]
+        run = run.drop_duplicates([index_col.SampleID, index_col.LaneNumber])
+        run[COL_LIBRARY] = run[index_col.SampleID].str.extract(
```

Comment:
Can you flag this with a TODO to fix it. We shouldn't be extracting libraries from file names or sample IDs.

Reply:
Done in 2252416. I thought this was an easy fix with Pinery Sample Provenance, but the 10X indices make it non-trivial. Not sure if qc-etl should be doing this.